quarkus/container/Dockerfile

```diff
@@ -1,36 +1,34 @@
+# Use Red Hat's UBI (Universal Base Image) as a build stage named 'ubi-micro-build'
 FROM registry.access.redhat.com/ubi9 AS ubi-micro-build
 
+# Set environment variable for Keycloak version and download link
 ENV KEYCLOAK_VERSION 999.0.0-SNAPSHOT
 ARG KEYCLOAK_DIST=https://github.com/keycloak/keycloak/releases/download/$KEYCLOAK_VERSION/keycloak-$KEYCLOAK_VERSION.tar.gz
 
-RUN dnf install -y tar gzip
-
-ADD $KEYCLOAK_DIST /tmp/keycloak/
-
-# The next step makes it uniform for local development and upstream built.
-# If it is a local tar archive then it is unpacked, if from remote is just downloaded.
-RUN (cd /tmp/keycloak && \
-    tar -xvf /tmp/keycloak/keycloak-*.tar.gz && \
-    rm /tmp/keycloak/keycloak-*.tar.gz) || true
-
-RUN mv /tmp/keycloak/keycloak-* /opt/keycloak && mkdir -p /opt/keycloak/data
-RUN chmod -R g+rwX /opt/keycloak
+# Install necessary packages, add and unpack Keycloak, then clean up in one layer
+RUN dnf install -y tar gzip && \
+    ADD $KEYCLOAK_DIST /tmp/keycloak/ && \
+    (cd /tmp/keycloak && tar -xvf /tmp/keycloak/keycloak-*.tar.gz && rm /tmp/keycloak/keycloak-*.tar.gz) || true && \
+    mv /tmp/keycloak/keycloak-* /opt/keycloak && \
+    mkdir -p /opt/keycloak/data && \
+    chmod -R g+rwX /opt/keycloak && \
+    dnf clean all
```

Comment:
Although normally command grouping into a single RUN instruction would be more optimal, it's actually not in this instance. This first build stage is discarded, so we can be as inefficient as we want with the layers. It'll be a tiny bit slower to build, but that's it

In my opinion, optimizing them in this context is a net negative, because it removes a lot of clarity from the code without the normal benefits

Reply:
Oh, and additionally `dnf clean all` is not required, for similar reasons